### PR TITLE
[Core] Improve the server startup error message.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Add a message with better action item. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/rpc/grpc_server.cc [hunk 1]
```diff
@@ -58,8 +58,12 @@ void GrpcServer::Run() {
   // Build and start server.
   server_ = builder.BuildAndStart();
 
-  RAY_CHECK(server_) << "Failed to start the grpc server. The specified port is "
-                     << specified_port;
+  RAY_CHECK(server_)
+      << "Failed to start the grpc server. The specified port is " << specified_port
```
Comment (richardliaw) [inline src/ray/rpc/grpc_server.cc:62]: what's the GRPC server?

Comment (richardliaw) [reply src/ray/rpc/grpc_server.cc:62]: asking as a ray user

Comment (rkooo567) [reply src/ray/rpc/grpc_server.cc:62]: Ah, good point. Let me try improving it.

Comment (rkooo567) [reply src/ray/rpc/grpc_server.cc:62]: Check again?

File: src/ray/rpc/grpc_server.cc [hunk 2]
```diff
@@ -58,8 +58,16 @@ void GrpcServer::Run() {
...
+      << ". Grpc servers are used for ray's core components such as raylet, gcs "
+         "sever, or worker processes. If the server startup error message is Address "
+         "already in use, this "
+         "indicates the server fails to start because the port is already used by "
+         "other processes. It means --node-manager-port, --object-manager-port, "
+         "--gcs-server-port, and ports between --min-worker-port, --max-worker-port are "
+         "already used. If you see this error message, please try lsof -i :[port_number] "
```
Comment (richardliaw) [inline src/ray/rpc/grpc_server.cc:69]:
```suggestion
         "other processes (such as --node-manager-port, --object-manager-port, "
         "--gcs-server-port, and ports between --min-worker-port, --max-worker-port). "
         " Try running `lsof -i :" << specified_port << "` "
```

File: src/ray/rpc/grpc_server.cc [hunk 3]
```diff
@@ -58,8 +58,16 @@ void GrpcServer::Run() {
   // Build and start server.
   server_ = builder.BuildAndStart();
 
-  RAY_CHECK(server_) << "Failed to start the grpc server. The specified port is "
-                     << specified_port;
+  RAY_CHECK(server_)
+      << "Failed to start the grpc server. The specified port is " << specified_port
+      << ". Grpc servers are used for ray's core components such as raylet, gcs "
+         "sever, or worker processes. If the server startup error message is Address "
+         "already in use, this "
```
Comment (richardliaw) [inline src/ray/rpc/grpc_server.cc:65]:
```suggestion
         "sever, or worker processes. If the server startup error message is `Address "
         "already in use`, this "
```

File: src/ray/rpc/grpc_server.cc [hunk 4]
```diff
@@ -58,8 +58,16 @@ void GrpcServer::Run() {
   // Build and start server.
   server_ = builder.BuildAndStart();
 
-  RAY_CHECK(server_) << "Failed to start the grpc server. The specified port is "
-                     << specified_port;
+  RAY_CHECK(server_)
+      << "Failed to start the grpc server. The specified port is " << specified_port
+      << ". Grpc servers are used for ray's core components such as raylet, gcs "
+         "sever, or worker processes. If the server startup error message is Address "
```
Comment (richardliaw) [inline src/ray/rpc/grpc_server.cc:64]: (we don't need to dive into too much detail here imo)
```suggestion
      << ". This means that Ray's core components will not be able to function correctly." If the server startup error message is Address "
```

**Review by richardliaw — APPROVED** [review]
nice!

**Pull request merged**